### PR TITLE
doc: Bump to Ubuntu Bionic 18.04 in build-windows.md

### PR DESCRIPTION
**Pull request by ken2812221**
Windows starts to provide Ubuntu 18.04 WSL, it can be downloaded from Microsoft Store.

### CODE REVIEW

**Comment by promag** [comment]
Concept ACK.

There are more references to 17.10 that could be bumped?

**Comment by fanquake** [comment]
Concept ACK
This will need a few more updates though. For example Ubuntu and Ubuntu 18.04 are two different items in the Windows Store, so we should link to 18.04 from here.

**Comment by MarcoFalke** [comment]
I had the impression that we only support windows cross builds for bionic, due to qt 5.9. So might as well remove all previous versions/workarounds?

**Comment by ken2812221** [comment]
According to @MarcoFalke's comment, removed all other distros except for Ubuntu 18.04. Also changed the Ubuntu 18.04 WSL link according to @fanquake's comment.

**Comment by fanquake** [comment]
Thanks @ken2812221 . Instructions are looking simpler now. Was going to run through a build but VirtualBox has crashed twice while building depends using 18.04 WSL.. Will ACK once I've got a build finished.

**Comment by ken2812221** [comment]
Update: Search "Ubuntu 18.04"

**Comment by laanwj** [comment]
Concept ACK. I like this as it gets rid of a lot of confusing backwards compatibility cruft.

File: doc/build-windows.md [hunk 1]
```diff
@@ -38,10 +38,10 @@ To install WSL on Windows 10 with Fall Creators Update installed (version >= 162
   * Enable 'Windows Subsystem for Linux'
   * Click 'OK' and restart if necessary
 2. Install Ubuntu
-  * Open Microsoft Store and search for Ubuntu or use [this link](https://www.microsoft.com/store/productId/9NBLGGH4MSV6)
+  * Open Microsoft Store and search for "Ubuntu 18.04" or use [this link](https://www.microsoft.com/store/productId/9N9TNGVNDL3Q)
   * Click Install
 3. Complete Installation
-  * Open a cmd prompt and type "Ubuntu"
+  * Open a cmd prompt and type "Ubuntu1804"
```
Comment (laanwj) [inline doc/build-windows.md]: Will this stay this way? Or is this temporarily, until `1804` becomes the default

File: doc/build-windows.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Below are some notes on how to build Bitcoin Core for Windows.
 
 The options known to work for building Bitcoin Core on Windows are:
 
-* On Linux using the [Mingw-w64](https://mingw-w64.org/doku.php) cross compiler tool chain. Ubuntu Trusty 14.04 is recommended
+* On Linux using the [Mingw-w64](https://mingw-w64.org/doku.php) cross compiler tool chain. Ubuntu Bionic 18.04 is recommended
```
Comment (laanwj) [inline doc/build-windows.md]: Please add that these steps only apply for `18.04`. Only saying that it is recommended is not enough, as it means you'd still have to warn people against using `16.04`.

File: doc/build-windows.md (hunk 1 above)
Comment (ken2812221) [reply doc/build-windows.md]: I'm not sure, it's a different app in Microsoft Store now.

File: doc/build-windows.md (hunk 2 above)
Comment (ken2812221) [reply doc/build-windows.md]: Change `recommended` to `required`

File: doc/build-windows.md (hunk 1 above)
Comment (fanquake) [reply doc/build-windows.md]: There are multiple different Ubuntu "Apps" available in the Windows store, so I think this should be ok for now.

**Comment by fanquake** [comment]
TestedACK https://github.com/bitcoin/bitcoin/pull/13246/commits/9d4f9421a4898f75a4c49aaefaba2868de071d79

This is a good cleanup, less workarounds.

I've done a 64 bit depends build using WSL on a clean Windows 10 VM @ https://github.com/bitcoin/bitcoin/commit/6916024768ec57a00f54224640ab4e4871d2a30a
wsl

**Comment by MarcoFalke** [comment]
utACK 9d4f9421a4898f75a4c49aaefaba2868de071d79

**Pull request merged**